python/python/tests/test_async_client.py

```diff
@@ -1944,6 +1944,43 @@ async def test_zrank(self, redis_client: TRedisClient):
         with pytest.raises(RequestError):
             await redis_client.zrank(key, "one")
 
+    @pytest.mark.parametrize("cluster_mode", [True, False])
+    @pytest.mark.parametrize("protocol", [ProtocolVersion.RESP2, ProtocolVersion.RESP3])
+    async def test_zdiff(self, redis_client: TRedisClient):
+        key1 = f"{{testKey}}:1-{get_random_string(10)}"
+        key2 = f"{{testKey}}:2-{get_random_string(10)}"
+        key3 = f"{{testKey}}:3-{get_random_string(10)}"
+        string_key = f"{{testKey}}:4-{get_random_string(10)}"
+        non_existing_key = f"{{testKey}}:5-{get_random_string(10)}"
+
+        member_scores1 = {"one": 1.0, "two": 2.0, "three": 3.0}
+        member_scores2 = {"two": 2.0}
+        member_scores3 = {"one": 1.0, "two": 2.0, "three": 3.0, "four": 4.0}
+
+        assert await redis_client.zadd(key1, member_scores1) == 3
+        assert await redis_client.zadd(key2, member_scores2) == 1
+        assert await redis_client.zadd(key3, member_scores3) == 4
+
+        assert await redis_client.zdiff([key1, key2]) == ["one", "three"]
+        assert await redis_client.zdiff([key1, key3]) == []
+        assert await redis_client.zdiff([non_existing_key, key3]) == []
+
+        assert await redis_client.zdiff_withscores([key1, key2]) == {
+            "one": 1.0,
+            "three": 3.0,
+        }
+        assert await redis_client.zdiff_withscores([key1, key3]) == {}
+        assert await redis_client.zdiff_withscores([non_existing_key, key3]) == {}
+
```

Comment:
please add a test for cross-slot case

Reply:
Good point, note that this test is missing for the sorted set commands in Java so they will need to be updated